packages/flutter/lib/src/material/elevated_button.dart

```diff
@@ -316,6 +316,18 @@ class ElevatedButton extends ButtonStyleButton {
   ButtonStyle? themeStyleOf(BuildContext context) {
     return ElevatedButtonTheme.of(context).style;
   }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(ObjectFlagProperty<VoidCallback>('onPressed', onPressed, ifNull: 'disabled'));
```

Comment:
If ButtonStyleButton was defined `with Diagnosticable` then these properties could be defined there and the subclasses would just inherit them.

Reply:
Hey, thanks for taking a look.

I tried that before and tried it again today but when defining  `Diagnosticable` in the abstract `ButtonStyleButton` still fails the tests (`dev/integration_tests/flutter_gallery/test/smoke_test`) with the changes from https://github.com/flutter/flutter/pull/99753 but passes with the changes in this PR

Reply:
Were you able to figure out why adding it to `ButtonStyleButton` didn't work?  It seems like a better change to me to do as Hans suggested, but if it's not possible, it would be good to know why.

Reply:
After more digging and independent tests, I realize the problem is with the overlay MaterialStateProperty I've provided to ElevatedButton in the original PR , it returns an  "Instance" which failed this test below:

https://github.com/flutter/flutter/blob/65caffd4053718dd8e82aa069c11709370dee9cb/dev/integration_tests/flutter_gallery/test/smoke_test.dart#L49

This PR that fixed the problem, which misled me to believe this was the only solution.
I've fixed the problem in original PR that was blocked by this, I can close this now :)